package/com.unity.formats.usd/Tests/Editor/EditorAnalyticsBaseFixture.cs

```diff
@@ -0,0 +1,150 @@
+            EditorAnalytics.SendAnalyticsEventsImmediately = true;
+        }
+
+        [OneTimeTearDown]
+        public void RevertEditorAnalyticsSettings()
+        {
+            // Maybe implmeent a package clean up here
+
+            EditorAnalytics.enabled = m_initialAnalyticsSetting;
+            EditorAnalytics.recordEventsEnabled = m_initialRecordEventsSetting;
+            EditorAnalytics.SendAnalyticsEventsImmediately = m_initialSendEventsImmediatelySetting;
+        }
+
+        [SetUp]
+        public void ResetEventList()
+        {
+            DebuggerEventListHandler.ClearEventList();
+            InitUsd.Initialize();
+        }
+
+        public IEnumerator WaitForUsdAnalytics<T>(string expectedType, System.Action<AnalyticsEvent> expectedEvent, float attemptFrameCountLimit = 300) where T: UsdAnalyticsEvent
+        {
+            bool found = false;
+            float currFrameCount = 0;
+
+            while (!found && currFrameCount < attemptFrameCountLimit)
+            {
+                AnalyticsEvent expectedAnalyticsEvent = new AnalyticsEvent();
+                foreach (var concattedEvent in DebuggerEventListHandler.fetchEventList())
+                {
+                    if (!concattedEvent.Contains(expectedType))
+                    {
+                        continue;
+                    }
+
+                    Debug.Log($"Raw Log: {concattedEvent}");
+
+                    var splitEvents = concattedEvent.Split("\n");
+                    var usdIndex = FindUsdAnalyticsIndex(expectedType, splitEvents);
+
+                    //var sharedEvent = JsonConvert.DeserializeObject<SharedAnalyticsEvent>(splitEvents[0]);
+                    var usdEvent = JsonUtility.FromJson<T>(splitEvents[usdIndex]);
+
+                    if (usdEvent.type.Contains(expectedType))
+                    {
+                        //expectedAnalyticsEvent.shared = sharedEvent;
+                        expectedAnalyticsEvent.usd = usdEvent;
+
+                        expectedEvent(expectedAnalyticsEvent);
+                        found = true;
+                        break;
+                    }
+                }
+
+                DebuggerEventListHandler.ClearEventList();
+                currFrameCount++;
+
+                yield return null;
+            }
+        }
+
+        private int FindUsdAnalyticsIndex(string expectedType, string[] splitEvents)
+        {
+            for (int index = 0; index < splitEvents.Length; index++)
+            {
+                if (splitEvents[index].Contains(expectedType))
+                {
+                    return index;
+                }
+            }
+
+            return -1;
+        }
+    }
+
+    public class AnalyticsEvent
+    {
+        public SharedAnalyticsEvent shared;
+        public UsdAnalyticsEvent usd;
+    }
+
+    public class SharedAnalyticsEvent
+    {
+        // check if the class for this is in Unity source (if its a bit convoluted just make it)
```

Comment:
What does this comment mean?

Reply:
It can be removed, but i meant to check if theres a class with the same structure as "SharedAnalyticsEvent" already defined within Unity, and if so, use that instead of creating it here